### DISCUSSION TITLE
Yet Another Shape Question

### DISCUSSION THREAD

**Discussion by smokinguns** [OrchardCMS/OrchardCore#7030] [Q&A] [answered]
[Found this issue](https://github.com/OrchardCMS/OrchardCore/issues/6504#issue-644832493)

I too created something similar and wanted to inject into layout page.

{% shape "carousel", alias: "alias:hero-carousel"}

the carousel does render with Carsouel.liquid, but {{Model| json}} renders []. I would have expected the Model to contain the properties of my Carousel Content

In the end i just want guidance on how to create content pieces that I can reference in my layout page.

**Comment by Skrypt** [comment, marked as answer]
Look at this example to understand how to code a custom shape : https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Menu/MenuShapes.cs
This is how we create the menu shape on the Layout page of each theme.

Now if that's what you did ; then make sure you have registered it properly in your startup.cs of your custom project where your Carousel shape is defined so that it gets added to the IShapeTableProvider. Then also, of course, make sure you enabled that feature (module) in your admin.
